### PR TITLE
[Rust] rust lang initial version

### PR DESCRIPTION
**Pull request by wangweipeng2**
## What do these changes do?
Supporting Rust language. The protocol is not guaranteed to be aligned at the moment, and will continue to be improved in the future.

These types of serialization and deserialization have been implemented.
```rust
pub enum FieldType {
        STRING = 13,
	ARRAY = 25,
	MAP = 30,
	BOOL = 1,
	UINT8 = 2,
	INT8 = 3,
	UINT16 = 4,
	INT16 = 5,
	UINT32 = 6,
	INT32 = 7,
	UINT64 = 8,
	INT64 = 9,
	FLOAT = 11,
	DOUBLE = 12,
	BINARY = 14,
	DATE = 16,
	TIMESTAMP = 18,
        FuryTypeTag = 256,
	FurySet = 257,
	FuryPrimitiveBoolArray = 258,
	FuryPrimitiveShortArray = 259,
	FuryPrimitiveIntArray = 260,
	FuryPrimitiveLongArray = 261,
	FuryPrimitiveFloatArray = 262,
	FuryPrimitiveDoubleArray = 263,
	FuryStringArray = 264,
}
```

## Related issue number
#689

### CODE REVIEW

**Comment by chaokunyang** [comment]
@wangweipeng2 Awesome work, could you please add codelint to lint ci?

File: rust/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    "fury_derive",
+    "fury",
+    "test_suite"
+]
+exclude = [
+    "fuzz"
+]
+
+[workspace.package]
+version = "0.0.1"
```
Comment (chaokunyang) [inline rust/Cargo.toml:12]:
```suggestion
version = "0.1.0.dev"
```

File: rust/fury/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+[package]
+name = "fury"
+version = "0.1.0"
+edition = "2021"
```
Comment (chaokunyang) [inline rust/fury/Cargo.toml:4]:
```suggestion
edition = "2023"
```

**Comment by chaokunyang** [comment]
@wangweipeng2 Some filename abbr are too short, can we use a more complete name?

**Review by chaokunyang — APPROVED** [review]
LGTM

**Comment by wangweipeng2** [comment]
> @wangweipeng2 Some filename abbr are too short, can we use a more complete name?

OK，I will fix it together with the upcoming changes.

**Pull request merged**